common/numerated/src/interval.rs

```diff
@@ -85,352 +67,251 @@ impl<T: Numerated> NonEmptyInterval<T> {
-        NonEmptyInterval::new(T::min_value(), range.end)
-            .unwrap_or_else(|| {
-                unreachable!(
-                    "`T: LowerBounded` impl error: for any x: T must be T::min_value() ≤ x"
-                )
-            })
-            .into()
-    }
-}
-
-impl<T: Numerated + UpperBounded, I: Into<T::B>> From<RangeFrom<I>> for Interval<T> {
-    fn from(range: RangeFrom<I>) -> Self {
-        let start: T::B = range.start.into();
-        match start.unbound() {
-            BoundValue::Value(start) => NonEmptyInterval::new(start, T::max_value())
-                .unwrap_or_else(|| {
-                    unreachable!(
-                        "`T: UpperBounded` impl error: for any x: T must be x ≤ T::max_value()"
-                    )
-                })
-                .into(),
-            BoundValue::Upper(start) => Self { start, end: None },
-        }
+        let (start, end) = (T::min_value(), range.end);
+        debug_assert!(start <= end, "`T: LowerBounded` impl error");
+        Self { start, end }
     }
 }
 
 impl<T: Numerated + UpperBounded + LowerBounded> From<RangeFull> for Interval<T> {
     fn from(_: RangeFull) -> Self {
-        NonEmptyInterval::new(T::min_value(), T::max_value()).unwrap_or_else(|| {
-            unreachable!("`T: UpperBounded + LowerBounded` impl error: must be T::min_value() ≤ T::max_value()")
-        }).into()
-    }
-}
-
-impl<T: Numerated + LowerBounded + UpperBounded, I: Into<T::B>> From<RangeTo<I>> for Interval<T> {
-    fn from(range: RangeTo<I>) -> Self {
-        let end: T::B = range.end.into();
-        let start = T::min_value();
-        match end.unbound() {
-            BoundValue::Value(end) => {
-                debug_assert!(end >= T::min_value());
-                let end = end.dec_if_gt(T::min_value());
-                Self { start, end }
-            }
-            BoundValue::Upper(end) => Self {
-                start,
-                end: Some(end),
-            },
-        }
+        let (start, end) = (T::min_value(), T::max_value());
+        debug_assert!(start <= end, "`T: UpperBounded + LowerBounded` impl error");
+        Self { start, end }
     }
 }
 
-/// Error, which occurs, when trying to convert `start` > `end` into `Interval`.
+/// Trying to make empty interval.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct IntoIntervalError;
+pub struct EmptyRangeError;
 
-impl<T: Numerated, S: Into<T::B>, E: Into<T::B>> TryFrom<(S, E)> for Interval<T> {
-    type Error = IntoIntervalError;
-
-    fn try_from((start, end): (S, E)) -> Result<Self, Self::Error> {
-        use BoundValue::*;
+/// Trying to make interval from range where start > end.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct IncorrectRangeError;
+
+/// Trying to make interval from range where start > end or empty range.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, derive_more::From)]
+pub enum IncorrectOrEmptyRangeError {
```

Comment:
Remove underlying structs in favour of just enum variants

Reply:
done